Rewritten autoinstallation chapter of client configuration guide (#917)
uyuni-project/uyuni-docs: modules/client-configuration/pages/autoinst-unattended.adoc

[[autoinst-unattended]]
= Unattended Provisioning

> In order to install via PXE boot the operating system on the client without the need to select the distribution manually at boot time, the client must already be known to {productname}.

Reviewer: ```suggestion
To install an operating system using PXE boot without having to select the distribution manually at boot time, the client must already be known to {productname}.
```

This is still super passive, but it's getting better.

> You may associate a client's MAC address with a profile. This is possible via the {productname} {api}.

Reviewer: ```suggestion
You can associate a client's MAC address with a profile using the {productname} API.
```

> For information about this kind of associations, see xref:client-configuration:autoinst-unattended.adoc#associate-mac-to-profile[Associate a MAC Address to a Profile].

Reviewer: ```suggestion
For more information about these kinds of associations, see xref:client-configuration:autoinst-unattended.adoc#associate-mac-to-profile[].
```

> Alternatively, the "bare metal" functionality allows to register any new machine as soon as it connects to the local network, thanks to a generic PXE boot image.

Reviewer: ```suggestion
Alternatively, the bare metal functionality allows you to register any new machine as soon as it connects to the local network, using a generic PXE boot image.
```

You then go to {productname} {webui} and assign a profile to this machine.

> The next time the client boots, the operating system gets installed according to that profile. For details, see the *relevant section*.
> xref:client-configuration:autoinst-unattended.adoc#bare-metal[Bare Metal Provisioning].

Reviewer: ```suggestion
The next time the client boots, the operating system is installed according to that profile. 
For more information, see xref:client-configuration:autoinst-unattended.adoc#bare-metal[].
```

[[associate-mac-to-profile]]
== Associate a MAC Address to a Profile

You can use API calls to associate the MAC address of a client to the profile to install on it.

.Procedure: Select the Profile to Install on a Client

> . At the command prompt, use the [systemitem]``system.createSystemProfile`` API call.

Reviewer: ```suggestion
. On the {productname} Server, at the command prompt, as root, use the [systemitem]``system.createSystemProfile`` API call:
```

+
----
# spacecmd api -- --args '["systemname", {"hwAddress": "<hw_addr>", "hostname":"<name>"}]' system.createSystemProfile
----
  In this example, replace [literal]``<hw_addr>`` with a hardware address such as [literal]``00:25:22:71:e7:c6`` and [literal]``<name>`` with the name of your client.

> . In the {productname} {webui}, navigate to [guimenu]``Systems``, locate your new clients in the list, and click the client you want to provision.
> . Select the menu:Provisioning[Autoinstallation] tab.
> . Select the {ay} profile to use.

Reviewer: ```suggestion
. In the {productname} {webui}, navigate to [guimenu]``Systems``, select the client to provision, and go to the menu:Provisioning[Autoinstallation] tab.
. Select the {ay} profile to use.
```

> When you start the installation at a later time, the client's MAC address will allow Cobbler to know its name and then use the right profile.

Reviewer: ```suggestion
When you start the installation at a later time, the client's MAC address allows Cobbler to know the correct name and profile to use.
```

[NOTE]
====

> It is recommanded to not directly use the almost equivalent command [command]``cobbler system add``.

Reviewer: ```suggestion
Do not use the [command]``cobbler system add`` command.
```

Why?
